src/background/resource/sessionCacheService.ts

```diff
@@ -1,11 +1,17 @@
+type Entry = { timestamp: number; value: unknown };
+
 export class SessionCacheService {
-  private cache: Record<string, { timestamp: number; value: unknown }> = {};
+  private cache: Record<string, Entry | undefined> = {};
```

Comment:
Do we really need to store `undefined` values? It type checks successfully without `| undefined` btw

Reply:
That's the problem :) typescript assumes the value will always be retrieved by any key

Reply:
And we're not allowing to pass undefined when setting a value btw

Reply:
ah, alright, makes sense!